src/test/java/org/terasology/manualLabor/systems/ShearingSystemTest.java

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.manualLabor.systems;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.entity.internal.PojoEntityManager;
+import org.terasology.engine.rendering.assets.material.Material;
+import org.terasology.engine.rendering.assets.skeletalmesh.SkeletalMesh;
+import org.terasology.engine.rendering.logic.SkeletalMeshComponent;
+import org.terasology.gestalt.assets.management.AssetManager;
+
+import java.util.Optional;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ShearingSystemTest {
+    ShearingSystem shearingSystem;
+    private EntityRef entity;
+
+    /**
+     * Initialize the shearing system and entity to be used while testing. The entity is supposed to mock the animal entity.
+     */
+    @BeforeEach
+    public void setUp() {
+        shearingSystem = new ShearingSystem();
+        entity = new PojoEntityManager().create();
+        SkeletalMeshComponent skeletalMeshComponent = new SkeletalMeshComponent();
+        entity.saveComponent(skeletalMeshComponent);
+    }
+
+    /**
+     * Check whether switchPrefab() changes mesh and material attributes of SkeletalMeshComponent correctly
+     */
+    @Test
+    public void testSwitchPrefabAssetChange() {
+        SkeletalMesh expectedMesh = mock(SkeletalMesh.class);
+        Material expectedMaterial = mock(Material.class);
+        createMockAssetManager(expectedMesh, expectedMaterial);
+
+        shearingSystem.switchPrefab(entity, "testMesh", "testMaterial");
+        SkeletalMeshComponent skeletalMeshComponent = entity.getComponent(SkeletalMeshComponent.class);
+
+        Assertions.assertEquals(skeletalMeshComponent.mesh, expectedMesh);
+        Assertions.assertEquals(skeletalMeshComponent.material, expectedMaterial);
```

Comment:
suggestion: A big part of `switchPrefab` that isn't explicit in this test is that we expect the values to be _different_ before and after calling it.

You could make that clearer here by either
* explicitly initializing the properties of the component in `setUp` so it's clear what they were to start with, or
* adding an assertion before calling `switchPrefab`, so we know it wasn't _expectedMesh_ to start with, or
* adding an assertion after calling `switchPrefab` a second time with a different value.